### PR TITLE
CONSOLE-3883: Improve local plugin development docs

### PR DESCRIPTION
**Pull request by vojtechszocs**
- revisit the section on local plugin development
- add section on using local plugin SDK code
- add section on changes in Console provided shared modules

### CODE REVIEW

**Comment by vojtechszocs** [comment]
cc @opayne1

**Review by rhamilto — APPROVED** [review]
/lgtm

**Review by rhamilto — APPROVED** [review]
/lgtm

**Comment by rhamilto** [comment]
QE Approver:
/assign @yapei
Docs Approver:
/assign @opayne1
PX Approver:
/assign @RickJWagner

**Comment by yapei** [comment]
the doc changes look good to us
/label qe-approved

**Comment by RickJWagner** [comment]
Thank you for these docs improvements.  Docs are the public face of our products, very important for customers and Red Hatters alike.

/label px-approved

File: frontend/packages/console-dynamic-plugin-sdk/README.md [hunk 1]
```diff
@@ -283,18 +298,57 @@ initiated requests when the MIME type of requested asset is not valid.
...
-`window.SERVER_FLAGS.consolePlugins` to see the list of plugins which Console loads upon its startup.
+To work with multiple plugins, provide multiple arguments to Bridge server:
+
+```sh
+./bin/bridge \
+  -plugins foo-plugin=http://localhost:9001/ -i18n-namespaces=plugin__foo-plugin \
+  -plugins bar-plugin=http://localhost:9002/ -i18n-namespaces=plugin__bar-plugin
+```
+
+Once the Bridge server is running, start your plugin web server(s) and ensure that plugin assets can
```
Comment (opayne1) [inline frontend/packages/console-dynamic-plugin-sdk/README.md:324]:
```suggestion
Once the Bridge server is running, start your plugin web server(s), and ensure that plugin assets can
```

File: frontend/packages/console-dynamic-plugin-sdk/README.md [hunk 2]
```diff
@@ -283,18 +298,57 @@ initiated requests when the MIME type of requested asset is not valid.
...
+be fetched via `/api/plugins/<plugin-name>` Bridge endpoint. For example, the following URLs should
+provide the same content:
+
+- http://localhost:9000/api/plugins/foo-plugin/plugin-manifest.json
+- http://localhost:9001/plugin-manifest.json
+
+Open Console in web browser and inspect the value of `window.SERVER_FLAGS.consolePlugins` to see the
```
Comment (opayne1) [inline frontend/packages/console-dynamic-plugin-sdk/README.md:331]:
```suggestion
Open the Console in your web browser and inspect the value of `window.SERVER_FLAGS.consolePlugins` to see the
```

File: frontend/packages/console-dynamic-plugin-sdk/README.md [hunk 3]
```diff
@@ -283,18 +298,57 @@ initiated requests when the MIME type of requested asset is not valid.
...
+list of dynamic plugins which Console loads upon its startup. For local development, this should only
```
Comment (opayne1) [inline frontend/packages/console-dynamic-plugin-sdk/README.md:332]:
```suggestion
list of dynamic plugins the Console loads at runtime. For local development, this should only
```

File: frontend/packages/console-dynamic-plugin-sdk/README.md [hunk 4]
```diff
@@ -283,18 +298,57 @@ initiated requests when the MIME type of requested asset is not valid.
...
+include plugin(s) listed via `-plugins` Bridge argument.
+
+### Using local Console plugin SDK code
+
+If you need to make modifications to Console dynamic plugin SDK code and have them reflected in your
```
Comment (opayne1) [inline frontend/packages/console-dynamic-plugin-sdk/README.md:337]:
```suggestion
If you need to make modifications to Console dynamic plugin SDK code and reflect them in your
```

File: frontend/packages/console-dynamic-plugin-sdk/README.md [hunk 5]
```diff
@@ -283,18 +298,57 @@ initiated requests when the MIME type of requested asset is not valid.
...
+plugin builds, follow these steps:
+
+1. Make changes in Console repo. Run `yarn build` in `frontend/packages/console-dynamic-plugin-sdk`
+   directory to rebuild plugin SDK files at `frontend/packages/console-dynamic-plugin-sdk/dist`.
+2. Make sure your plugin's `package.json` dependencies refer to local plugin SDK files, for example:
+```json
+"@openshift-console/dynamic-plugin-sdk": "file:../openshift/console/frontend/packages/console-dynamic-plugin-sdk/dist/core",
+"@openshift-console/dynamic-plugin-sdk-webpack": "file:../openshift/console/frontend/packages/console-dynamic-plugin-sdk/dist/webpack",
+```
+3. Refresh your plugin's `node_modules` whenever the local plugin SDK files are changed:
```
Comment (opayne1) [inline frontend/packages/console-dynamic-plugin-sdk/README.md:347]:
```suggestion
3. Refresh your plugin's `node_modules` whenever you change local plugin SDK files:
```

**Review by opayne1 — COMMENTED** [review]
A few suggestions :)

**Comment by opayne1** [comment]
Sorry I missed this, feel free to ping me when you are ready for the docs-approved label.

**Comment by vojtechszocs** [comment]
@opayne1 Thank you for the review! I have applied your suggestions.

**Comment by opayne1** [comment]
Thank you!

/label docs-approved

**Pull request merged**